src/ert/_c_wrappers/enkf/analysis_config.py

```diff
@@ -61,42 +42,46 @@ def __init__(
         self._analysis_iter_config = analysis_iter_config or AnalysisIterConfig()
         self._update_log_path = update_log_path
 
-        self._analysis_select = analysis_select
-        self._analysis_set_var = analysis_set_var or []
-        self._analysis_copy = analysis_copy or []
+        analysis_set_var = analysis_set_var or []
+        analysis_copy = analysis_copy or []
+        es_module = AnalysisModule.ens_smoother_module()
+        ies_module = AnalysisModule.iterated_ens_smoother_module()
+        self._modules: Dict[str, AnalysisModule] = {
+            AnalysisMode.ENSEMBLE_SMOOTHER: es_module,
+            AnalysisMode.ITERATED_ENSEMBLE_SMOOTHER: ies_module,
+        }
+        self._active_module = analysis_select or AnalysisMode.ENSEMBLE_SMOOTHER
+        # copy modules
+        for element in analysis_copy:
+            if isinstance(element, list):
+                src_name, dst_name = element
+            else:
+                src_name = element[ConfigKeys.SRC_NAME]
+                dst_name = element[ConfigKeys.DST_NAME]
 
-        if c_ptr:
-            super().__init__(c_ptr)
-            # copy modules
-            for element in self._analysis_copy:
-                if isinstance(element, list):
-                    src_name, dst_name = element
+            module = self._modules.get(src_name)
+            if module is not None:
+                if module.mode == AnalysisMode.ENSEMBLE_SMOOTHER:
+                    new_module = AnalysisModule.ens_smoother_module(dst_name)
                 else:
-                    src_name = element[ConfigKeys.SRC_NAME]
-                    dst_name = element[ConfigKeys.DST_NAME]
-
-                self._add_module_copy(
-                    src_name,
-                    dst_name,
+                    new_module = AnalysisModule.iterated_ens_smoother_module(dst_name)
+                self._modules[dst_name] = new_module
+            else:
+                raise AnalysisConfigError(
+                    f"Trying to copy module {src_name}" f" which does not exist"
                 )
 
-            # set var list
-            for set_var in self._analysis_set_var:
-                if isinstance(set_var, list):
-                    module_name, var_name, value = set_var
-                else:
-                    module_name = set_var[ConfigKeys.MODULE_NAME]
-                    var_name = set_var[ConfigKeys.VAR_NAME]
-                    value = set_var[ConfigKeys.VALUE]
-
-                module = self._get_module(module_name)
-                module._set_var(var_name, str(value))
-
-            if self._analysis_select:
-                self._select_module(self._analysis_select)
+        # set var list
+        for set_var in analysis_set_var:
```

Comment:
if previous comment is a good idea - maybe also here call a `_set_var_list()`  function?